application/ui/src/features/inspect/inference-provider.component.tsx

```diff
@@ -32,7 +30,11 @@ const downloadImageAsFile = async (media: MediaItem) => {
 
 const useInferenceMutation = () => {
     const pipeline = usePipeline();
-    const inferenceMutation = $api.useMutation('post', '/api/projects/{project_id}/models/{model_id}:predict');
+    const inferenceMutation = $api.useMutation('post', '/api/projects/{project_id}/models/{model_id}:predict', {
```

Comment:
For a follow up PR, would it be possible to change `mutation` to be a `query`? `$api` might not support this out of the box but we could maybe write our own, where we take a query key based project id and the media's id?
Then you might not need `resetInference`.

Then again this could also over complicate things so feel free to ignore the comment.

Reply:
I tried doing that, but I couldn’t get it working easily and got frustrated after spending half the day with no progress. So I went with this approach for now, I’ll revisit it in my next PR.